src-self-hosted/codegen.zig

```diff
@@ -773,8 +773,8 @@ fn Function(comptime arch: std.Target.Cpu.Arch) type {
                 .embedded_in_code => {
                     return self.fail(inst.base.src, "TODO implement storing to MCValue.embedded_in_code", .{});
                 },
-                .register => {
-                    return self.fail(inst.base.src, "TODO implement storing to MCValue.register", .{});
+                .register => |reg| {
+                    try self.setRegOrMem(inst.base.src, elem_ty, .{ .register = reg }, value);
```

Comment:
The semantics here are not quite right- this should behave as if the register has a pointer address in it, and store to that address. This is why I think to accomplish removing the operand from the return statement we need a "Set the return value" instruction, which would not necessarily operate on a pointer.